File: 실습/3장/08_local_function/Main.java
public class Main {

    public static void printUser(String name, int age) {
        validate(name, age);
        System.out.println(name + ", " + age);
    }

    private static void validate(String name, int age) {
        if (name.isBlank()) {
            throw new IllegalArgumentException(
                    "이름은 비어 있을 수 없습니다."
            );
        }

        if (age < 0) {
            throw new IllegalArgumentException(
                    "나이는 음수일 수 없습니다."
            );
        }
    }

    public static void main(String[] args) {
        printUser("철수", 20);
    }
}

File: 실습/3장/08_local_function/Main.kt
fun printUser(name: String, age: Int) {
    fun validate() {
        require(name.isNotBlank()) {
            "이름은 비어 있을 수 없습니다."
        }

        require(age >= 0) {
            "나이는 음수일 수 없습니다."
        }
    }

    validate()
    println("$name, $age")
}

fun main() {
    printUser("철수", 20)
}